src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -56,67 +56,72 @@
-    HINSTANCE shCoreDll = LoadLibrary(L"Shcore.dll");
-
-    if (shCoreDll)
+    for (int i = 0; i < maxTouchPoints; ++i)
     {
-        enum ProcessDpiAwareness
-        {
-            ProcessDpiUnaware         = 0,
-            ProcessSystemDpiAware     = 1,
-            ProcessPerMonitorDpiAware = 2
-        };
-
-        using SetProcessDpiAwarenessFuncType = HRESULT(WINAPI*)(ProcessDpiAwareness);
-        auto SetProcessDpiAwarenessFunc      = reinterpret_cast<SetProcessDpiAwarenessFuncType>(
-            reinterpret_cast<void*>(GetProcAddress(shCoreDll, "SetProcessDpiAwareness")));
-
-        if (SetProcessDpiAwarenessFunc)
+        if (touchIDs[i] == id)
+            return i;
+    }
+    for (int i = 0; i < maxTouchPoints; ++i)
+    {
+        if (touchIDs[i] == static_cast<DWORD>(-1))
         {
-            // We only check for E_INVALIDARG because we would get
-            // E_ACCESSDENIED if the DPI was already set previously
-            // and S_OK means the call was successful
-            if (SetProcessDpiAwarenessFunc(ProcessSystemDpiAware) == E_INVALIDARG)
-            {
-                sf::err() << "Failed to set process DPI awareness" << std::endl;
-            }
-            else
-            {
-                FreeLibrary(shCoreDll);
-                return;
-            }
+            touchIDs[i] = id;
+            return i;
         }
-
-        FreeLibrary(shCoreDll);
     }
+    return -1;
+}
 
-    // Fall back to SetProcessDPIAware if SetProcessDpiAwareness
-    // is not available on this system
-    HINSTANCE user32Dll = LoadLibrary(L"user32.dll");
+// Get a system error string from an error code
+std::string getErrorString(DWORD error)
+{
+    PTCHAR buffer;
+
+    if (FormatMessage(FORMAT_MESSAGE_MAX_WIDTH_MASK | FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM,
+                      nullptr,
+                      error,
+                      0,
+                      reinterpret_cast<PTCHAR>(&buffer),
+                      0,
+                      nullptr) == 0)
+        return "Unknown error.";
+
+    const sf::String message = buffer;
+    LocalFree(buffer);
+    return message.toAnsiString();
+}
 
-    if (user32Dll)
-    {
-        using SetProcessDPIAwareFuncType = BOOL(WINAPI*)();
-        auto SetProcessDPIAwareFunc      = reinterpret_cast<SetProcessDPIAwareFuncType>(
-            reinterpret_cast<void*>(GetProcAddress(user32Dll, "SetProcessDPIAware")));
+const GUID GUID_DEVINTERFACE_HID = {0x4d1e55b2, 0xf16f, 0x11cf, {0x88, 0xcb, 0x00, 0x11, 0x11, 0x00, 0x00, 0x30}};
 
-        if (SetProcessDPIAwareFunc)
-        {
-            if (!SetProcessDPIAwareFunc())
-                sf::err() << "Failed to set process DPI awareness" << std::endl;
-        }
+void setProcessDpiAware()
+{
+#if 0
+    // Windows 8.1+
+    if (SetProcessDpiAwareness(PROCESS_SYSTEM_DPI_AWARE) == E_INVALIDARG)
+    {
+        sf::err() << "Failed to set process DPI awareness" << std::endl;
+    }
+#endif
```

Comment:
What's the purpose of this inaccessible code?

Reply:
Should have added a comment there, it's mostly in there since I wasn't sure for now.

This would be the more modern/newer approach (and I think the specific behavior is slightly different), but would we want Windows 7 or Windows 8.1 as the minimum required/supported version?

Reply:
> but would we want Windows 7 or Windows 8.1 as the minimum required/supported version?

I say 8.1. SFML 3 needs to be forward-looking at Windows 7 has been out of fashion for a while now. I don't see the merit in supporting it. Can you even get a Visual Studio toolchain on W7 that supports C++17?